### PR TITLE
Fix flakiness in stream tests

### PR DESCRIPTION
**Pull request by afcapel**
These tests were a bit flaky because they were asserting DOM changes that can happen asynchronously.

Changing the assertions so we wait until the change happens fixes the flakiness.

Also, introduce `waitUntilText` to wait until some text appears in the page.

### CODE REVIEW

File: src/tests/helpers/page.ts [hunk 1]
```diff
@@ -282,6 +282,10 @@ export function waitForPathname(page: Page, pathname: string): Promise<void> {
   return page.waitForURL((url) => url.pathname == pathname)
 }
 
+export function waitUntilText(page: Page, text: string, state: "visible" | "attached" = "visible") {
```
Comment (kevinmcconnell) [inline src/tests/helpers/page.ts:285]: 👏

**Review by kevinmcconnell — APPROVED** [review]
Looks good!

**Comment by marcoroth** [comment]
Hey @afcapel and @kevinmcconnell, any chance you can rebase or squash merge the pull requests? I feel like the commit history otherwise gets really hard to parse and follow. Thank you!
